phase_teach.rb

```diff
@@ -0,0 +1,37 @@
+module PhaseTeach
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def teach_stuff
+    response = ""
+    if self.class == Teacher
+      response += "Listen, class, this is how everything works, fo SHO! "
+      response += "*drops flat-out insane knowledge bomb* "
+      response += "... You're welcome. *saunters away*"
+      response
+    elsif self.class == ApprenticeTeacher
+      response += "Listen, class, this is how everything works. "
+      response += "*drops crazy knowledge bomb* "
+      response += "... You're welcome."
+      response
+    end
+  end
+
+  def set_performance_rating(rating)
+    response = ""
+    if rating > 80 && self.class == ApprenticeTeacher
+      response = "Yay, I'm a great employee!"
+      receive_raise(@target_raise)
+    elsif rating > 90 && self.class == Teacher
```

Comment:
Also, why a module? If you have two classes that have a lot of the same code, your default should be to use inheritance if that makes sense (either one inherits from the other, or you create a common superclass), then if there's no sensible inheritance relationship (e.g. they both already inherit from other things, or they are just really inherently different kinds of things), use a module.